inlineplz/linter_runner.py

```diff
@@ -0,0 +1,332 @@
+            "stderr": asyncio.subprocess.PIPE,
+            "env": os.environ
+        }
+        # TODO: Figure out why we don't always get utf-8 back, even on 3.6.X without universal_newlines=True
+        if sys.version_info[0] >= 3 and sys.version_info[1] >= 6:
+            popen_kwargs["encoding"] = "utf-8"
+
+        # TODO: Figure out the max buffer size for subprocesses
+        async with self.process_limiter:
+            if sys.platform == "win32":
+                proc = await asyncio.wait_for(
+                    asyncio.create_subprocess_shell(" ".join(command), **popen_kwargs), timeout)
+            else:
+                proc = await asyncio.wait_for(
+                    asyncio.create_subprocess_exec(*command, **popen_kwargs), timeout)
+
+            stdout, stderr = await proc.communicate()
+
+        output = "{}\n{}".format(
+            stdout.decode("utf-8", "replace"), stderr.decode("utf-8", "replace")).strip()
+        return proc.returncode, output
+
+    async def performance_hacks(self):
+        # https://github.com/npm/npm/issues/11283
+        # npm's progress bar makes npm installs much slower
+        try:
+            await self.run_command(["npm", "set", "progress=false"])
+        except Exception:
+            pass
+
+    @staticmethod
+    def cleanup():
+        """Delete standard installation directories."""
+        for install_dir in inlineplz.linters.INSTALL_DIRS:
+            try:
+                shutil.rmtree(install_dir, ignore_errors=True)
+            except Exception:
+                print(traceback.format_exc())
+                print("Failed to delete {}".format(install_dir))
+
+    def should_ignore_path(self, path):
+        for ignore_path in self.ignore_paths:
+            if (
+                os.path.relpath(path).startswith(ignore_path)
+                or path.startswith(ignore_path)
+                or fnmatch.fnmatch(path, ignore_path)
+                or ignore_path in path.split(os.path.sep)
+            ):
+                return True
+
+        return False
+
+    async def run_per_file(self, config, path=None):
+        path = path or os.getcwd()
+        cmd = self.run_config(config)
+        cmds_and_tasks = []
+
+        paths = self.all_filenames_in_dir(path=path)
+        for pattern in registry.PATTERNS.get(config.get("language")):
+            for filepath in fnmatch.filter(paths, pattern):
+                if "text" in identify.tags_from_path(filepath):
+                    cmds_and_tasks.append((
+                        cmd + [filepath],
+                        self.event_loop.create_task(self.run_command(cmd + [filepath], timeout=5))
+                    ))
+
+        output = []
+        for run_cmd, file_task in cmds_and_tasks:
+            _, out = await file_task
+
+            # Return the filename and the output
+            output.append((run_cmd[-1], out.strip()))
+
+        return output
+
+    def linters_to_run(self):
```

Comment:
[inline-plz]: `prospector: mccabe: LinterRunner.linters_to_run is too complex (16) (MC0001)`

Reply:
i wonder why these dupes still happen